server/app/controllers/receive.py

```diff
@@ -57,7 +58,19 @@ class RequestArguments(TypedDict):
     'tca': fields.DateTime(required=True),
     'los': fields.DateTime(required=True),
     'sat': fields.Str(required=True),
-    'notes': fields.Str(required=False)
+    'notes': fields.Str(required=False),
+    'tle': fields.List(fields.Str, required=False,
+        validate=[
+            # TLE must contains exact two lines
+            validate.Length(equal=2, error="Require 2 line TLE. Actual: {input}"),
+            # TLE lines may be ended with trailling character
+            lambda tle: validate.Length(min=69, max=70,
```

Comment:
Length is just one of many properties that could be checked here. Take a look at tletools package. They have tons of checks there. But I think none of them are necessary here. If the TLE is broken, we should accept the observation anyway.

Reply:
I don't want to have invalid data in my DB.
I think that the most often errors will be empty string as TLE, TLE with additional line and TLE with missing line. This check validate all of this errors.
This solution provide fast and descriptive way to inform user what is wrong with request.
I right that it doesn't protect us from invalid TLE. If we add any parser for TLE in future then we may extend this check.
Now it checks a programming mistakes. I think that TLE will be not logically invalid if user used it to predict pass. Prediction software doesn't allow him to use incorrect TLE.